### ISSUE TITLE
Enhancement: Sharing whole project instead of current file

### ISSUE THREAD

**Issue by pxlrbt** [atom/teletype#211]
### Description

Current implementation only shares current project. It would be more useful if I could share the whole project so two persons can work on different files. Ideally jumping to the file by clicking an avatar.

Is anything like this planned for the future?

**Comment by alrz** [comment]
Had a similar idea, something like "code conference": basically, allows you to share a read-only workspace to others. Would like to know if this is practical. Thanks.

**Comment by pixelarbeit** [comment]
I was not only thinking about a "read-only" workspace, but also the possibility to edit other files. 
But maybe more fine-grained control would be interesting like: Read/Write; One File/Multiple Files/Whole project.

**Comment by VariableClass** [comment]
👍

**Comment by dennisvandehoef** [comment]
A colleague and I tested it, and it was the first thing we mentioned as "needed to be useful", but we also agreed that it would be enough to "only" share the open tabs. I mean, I have the same code base on my computer, so i can use that to find out which extra file is needed, and I can ask him to open it. In case security might be a big issue against it. ;)

**Comment by alrz** [comment]
@dennisvandehoef Good point. But it wouldn't be the case if the workspace is read-only. The tabs being changed should be distinguishable anyways, but at the same time you should be able to explore other files if you want. This is the desirable setting for a "code conference" use case.

**Comment by pixelarbeit** [comment]
@dennisvandehoef Thats why it would be good to have finer control. So you can set up read-only or write only to certain files. So you can choose your own security level.

For example a use case for browsing the whole project would be helping someone with his project, or looking for some bugs, so you down have to setup the code yourself. Would make it easier than setting up a Codepen every time.

**Comment by jasonrudolph** [comment]
Thanks for taking the time to share this idea! :bow: Repository-wide sharing and/or multi-file sharing is indeed something we'd like to provide in the future.

As we prioritize future work, we'll be looking at feedback issues like this one to determine where we can deliver the most benefit with the available engineering resources (i.e., full-time Atom engineers, community maintainers, and new contributors). In the meantime, if this is a feature that would bring you joy in your daily workflow, [adding a :+1: emoji reaction](https://github.com/blog/2119) to the [issue body](https://github.com/atom/teletype/issues/211#issue-274464843) is a great way to let us know.

**Comment by jasonrudolph** [comment]
:wave: Good news for people interested in collaborating on multiple files simultaneously: **Teletype v0.3.0 adds preliminary support for hosts to share multiple simultaneous files with guests**. Guests are free to navigate among the files shared by the host, and guests can edit those files independently of the file that the host is currently viewing/editing.

[(demo gif)](https://user-images.githubusercontent.com/2988/33579126-b2070d38-d916-11e7-92f9-bf67a3b73db5.gif)

This update is the first step toward implementing [RFC-001](https://github.com/atom/teletype/blob/2e369247ad645d982ab653b2fe1fce6ab5f52e61/doc/rfcs/001-allow-guests-to-open-multiple-remote-buffers.md). Note that RFC-001 focuses on the host sharing their workspace, which is a bit different than the host sharing an entire project. (See [*Rational and alternatives*](https://github.com/atom/teletype/blob/2e369247ad645d982ab653b2fe1fce6ab5f52e61/doc/rfcs/001-allow-guests-to-open-multiple-remote-buffers.md#rationale-and-alternatives) in RFC-001). We're interested in Teletype allowing a host to share an entire project at some point in the future. In the meantime though, we anticipate that sharing a workspace will address many of the use cases that developers have in mind when wanting to share an entire project.

If you're interested in tracking the progress on RFC-001, feel free to subscribe to the [tracking issue](https://github.com/atom/teletype/issues/268) for updates. 👌

**Comment by 0n11san** [comment]
jasonrudolph, is this update part of the main release now or is it a beta we have to dl ?

**Comment by jasonrudolph** [comment]
@0n11san: All Teletype releases are currently considered to be "beta" releases. This update is included in Teletype v0.3.0 and newer. If you already have Teletype installed, Atom should prompt you to update to the latest version (currently v0.5.0).

**Comment by Svahnen** [comment]
@jasonrudolph A better version of this would be to see the whole file structure as explained in https://github.com/atom/teletype/issues/227
The current version (0.9.0) allows ppl to work on multiple files, but it is very hard to navigate, especially when the host have files opened from different projects at once. What makes it even harder in the current state is that the clients will have the host name on each tab displayed as "@HostName SuperlongFilename.js" witch is cut down to "@HostName Su" and then have no clue what file that tab represents until opening it.

**Comment by AGkitchen** [comment]
Hi. After testing Teletype a bit and looking into this issue/reading the above comments, here's my 2 squid:
The most important way for me to be able to share is definitely to share the PROJECT FOLDER in its entirety. There is little utility in having to ping someone to nav over to a file so you can see it again, while hoping they never timeout. While sharing many files over Telemetry, any problem means I need to go click a bunch of tabs in a row. 
So the best way to handle security/selective sharing in this case it seems would be to use something like a GITIGNORE, and just list any patterns or filters for who you'd like to not see what.

**Comment by sancarn** [comment]
@AGkitchen 
Before a full permission system (like GITIGNORE) is setup, it makes sense to me to that a simple stepping stone would be to simply allow the guest to see the project tree, and request that files should be opened.

Later a role system could be implemented on top of this base functionality.

**Comment by ameliaikeda** [comment]
Been using teletype extensively at work with other devs on a new project, and while it's _fantastic_, the thing we'd really need in order to be even nearly as productive as in-person pairing is having access to the whole project. We're working on golang projects, and the ability to quickly jump between packages is ridiculously important!

**Comment by tobykas** [comment]
please, make it

**Comment by webtheo** [comment]
Is there any update? I think this is very important!

**Comment by BlueskyFR** [comment]
Being able to share en entire project is very important AND dangerous. When the "share" button is presses, the user should be prompt whether he wants to share current file or the entire project.
This means that git only becomes a way to have a backup as all the work is done using P2P!
This would simply be amazing.

Any update on this by the way?

**Comment by naturelover123321** [comment]
Will there be a function to permanently share the file to editors, similar to the way Google Drive works? This way, the host does not have to be online for guests to edit. Is there another way to do this? 

Thanks!

**Comment by mikefrederick** [comment]
Unfortunately this has myself and my team using VSCode + VS Live Share as we pair quite heavily. I miss atom :(. Do you need help contributing to this feature? Is there a way we can be given some direction or thoughts as to the route that might be taken?

**Comment by scoutchorton** [comment]
> Will there be a function to permanently share the file to editors, similar to the way Google Drive works? This way, the host does not have to be online for guests to edit. Is there another way to do this?
> 
> Thanks!

I would like this. I think possibly having a server version of Teletype would be a good implementation. Possibly have a backend of SFTP, have some handshake to connect to a project on a server, and have all the regular teletype features from there.

Me and a friend are working on a project, but we have to push to Git when we're dealing with programming flaws. Personally, I don't feel like Git should be used this way with many, many commits, but maybe I'm wrong! It would be nice to share an entire folder (like Google Drive), and be able to simultaneously edit the project.

Interesting idea to keep in mind...

**Comment by dhdbtkd** [comment]
It would be nice to share an entire folder (like Google Drive), and be able to simultaneously edit the project.

**Comment by bnebeker** [comment]
Is it on the roadmap to have the project view on the sidebar as well, or just the ability for the host to open files and the guest to see only the opened files in new tabs? I can see his code when he opens it, but my project view is blank. Would be helpful to have access to all files in the project, with some sort of host approval of course.

cc @jmwoloso

![image](https://user-images.githubusercontent.com/33212506/84956854-d4d8d680-b0ae-11ea-97d5-6897c2e67d2b.png)

**Comment by BlueskyFR** [comment]
The project seems almost abandoned. You can take a look at VSCode's Liveshare which works really well.
Moreover (and sadly), Atom generally almost doesn't receive any new update while VSCode does.

**Comment by bnebeker** [comment]
Dang - not a fan of VSCode but will give it a shot. Thanks!

**Comment by BlueskyFR** [comment]
> Dang - not a fan of VSCode but will give it a shot. Thanks!

I was thinking the same thing at the beginning too! But you can install the Atom keybindings extension on VSCode for example, and there is way more extensions (and more maintained!) as well!
I strongly advise anyone reading this to give VSCode a try as Atom stays in its current state (not gaining any new features).

**Comment by sancarn** [comment]
Hey @pxlrbt is there any reason why you closed this issue? If you're getting annoyed about notifications on the thread you can unsubscribe from the issue itself, instead of closing it...

**Comment by BlueskyFR** [comment]
Yes this issue is still unresolved! Why are you closing it?

**Comment by pxlrbt** [comment]
Sorry it seemed as this won't be implemented. But you're right. Unsubscribing is the better solution.
